webapp/app/forms/fields.py

```diff
@@ -352,6 +354,50 @@ def post_validate(self, form, validation_stopped):
             self.data = ''.join(self.data)
 
 
+class TaxNumberField(SteuerlotseStringField):
+    """
+        Field to store the tax number in either one or three separate input fields.
+
+        We get the formdata either as a list of one or three  strings (e.g. ['043', '4523', '3397'])
+        but want to handle it in the rest of the program as one string (e.g. '04345233397').
+        At the same time, in case of a validation error (e.g. for ['04', '4523', '3397']) we want to keep the order
+        of inputs to not confuse the user. Thus, we only concatenate the strings to one string on succeeded validation
+        in post_validate().
+        Once the input validates, we can be sure to have the complete, valid string in our data and
+        can split it into the expected chunks as seen in _value().
+    """
+    def process_formdata(self, valuelist):
+        if valuelist:
+            self.data = valuelist
+        elif self.data is None:
+            self.data = ''
```

Comment:
I have just checked it. An empty list leads to an error with the form data because that expects a string. I did not want to change that part that much because that leads to a lot of problems. I have just tested it and I do not see a problem that could be caused by this: a) because we normally do get a valuelist with empty strings. b) because if there is no valuelist and `self.data = ''` `value()` will return an empty array which is fine, too.

Nevertheless, maybe we can go through that part together?